bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/utils/DynamicDeckUtils.java

```diff
@@ -0,0 +1,340 @@
+                continue;
+            }
+
+            collectionMap.put(resultCollectionResource.getValueMap().get(JcrConstants.JCR_TITLE, String.class), resultCollectionResource.getPath());
+
+        }
+        return collectionMap;
+    }
+
+    /**
+     * This method will return asset Input stream for indd file
+     *
+     * @param resourceResolver
+     * @param assetResource
+     * @return
+     * @throws DynamicDeckDynamoException
+     */
+    public static InputStream getInddXmlRenditionInputStream(ResourceResolver resourceResolver,
+                                                             Resource assetResource) throws DynamicDeckDynamoException {
+        Resource assetJcrContentResource = resourceResolver.getResource(assetResource.getPath() + DynamicDeckDynamoConstants.SLASH
+                + JcrConstants.JCR_CONTENT + DynamicDeckDynamoConstants.SLASH + DamConstants.RENDITIONS_FOLDER + DynamicDeckDynamoConstants.SLASH
+                + StringUtils.replace(assetResource.getName(), DynamicDeckDynamoConstants.DOT + DynamicDeckDynamoConstants.INDD_EXTENSION,
+                DynamicDeckDynamoConstants.DOT + DynamicDeckDynamoConstants.XML_EXTENSION)
+                + DynamicDeckDynamoConstants.SLASH + JcrConstants.JCR_CONTENT);
+        return getInputStreamByResource(assetJcrContentResource);
+    }
+
+    public static InputStream getAssetInputStreamByPath(ResourceResolver resourceResolver, String annotatedXmlPath) throws DynamicDeckDynamoException {
+        Resource assetJcrContentResource = resourceResolver
+                .getResource(annotatedXmlPath + DynamicDeckDynamoConstants.SLASH + JcrConstants.JCR_CONTENT + DynamicDeckDynamoConstants.SLASH
+                        + DamConstants.RENDITIONS_FOLDER + DynamicDeckDynamoConstants.SLASH + DamConstants.ORIGINAL_FILE + DynamicDeckDynamoConstants.SLASH + JcrConstants.JCR_CONTENT);
+        return getInputStreamByResource(assetJcrContentResource);
+    }
+
+    private static InputStream getInputStreamByResource(Resource assetJcrContentResource) throws DynamicDeckDynamoException {
+        if (null == assetJcrContentResource) {
+            throw new DynamicDeckDynamoException("Annotated XML resource is null");
+        }
+        Node node = assetJcrContentResource.adaptTo(Node.class);
+        if (null == node) {
+            throw new DynamicDeckDynamoException("Annotated XML node is null, asset resource path: " + assetJcrContentResource.getPath());
+        }
+        try {
+            return node.getProperty(JcrConstants.JCR_DATA).getBinary().getStream();
+        } catch (RepositoryException e) {
+            throw new DynamicDeckDynamoException("Repository exception occurred while fetching the xml input stream", e);
+        }
+    }
+
+    /**
+     * This method commit resourceResolver object after session refresh
+     *
+     * @param resourceResolver
+     */
+    public static void commit(ResourceResolver resourceResolver) {
```

Comment:
In this method you are doing the commit on both the ResourceResolver and the JCR Session wrapped with the RR. Please just stick to the RR and omit doing the same for the JCR session here, it's not necessary (because the RR is doing that internally already).